kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -94,16 +115,31 @@ public LogSegment(
       checkArgument(!deltas.isEmpty() || !checkpoints.isEmpty(), "No files to read");
 
       if (!deltas.isEmpty()) {
+        final List<Long> deltaVersions =
+            deltas.stream()
+                .map(fs -> FileNames.deltaVersion(new Path(fs.getPath())))
+                .collect(Collectors.toList());
+
+        // Check the first delta version
         this.checkpointVersionOpt.ifPresent(
             checkpointVersion -> {
               checkArgument(
-                  FileNames.deltaVersion(deltas.get(0).getPath()) == checkpointVersion + 1,
+                  deltaVersions.get(0) == checkpointVersion + 1,
                   "First delta file version must equal checkpointVersion + 1");
             });
 
+        // Check the last delta version
         checkArgument(
-            FileNames.deltaVersion(ListUtils.getLast(deltas).getPath()) == version,
+            ListUtils.getLast(deltaVersions) == version,
             "Last delta file version must equal the version of this LogSegment");
+
+        // Ensure the delta versions are contiguous
+        for (int i = 1; i < deltaVersions.size(); i++) {
+          if (deltaVersions.get(i) != deltaVersions.get(i - 1) + 1) {
+            throw new IllegalArgumentException(
+                String.format("Delta versions must be contiguous: %s", deltaVersions));
```

Comment:
Can we use the same check we use in `SnapshotManager`?

Reply:
I considered that, yeah. My only intuition _not_ to use it was: this LogSegment will eventually be public API. So if the engine constructs and invalid LogSegment to pass in -- do we really want to throw an InvalidTableException, or would IllegalArgumentException be more appropriate? Open to your thoughts.

Reply:
yeah okay that makes sense to me. I think IllegalArgumentException is much better here!